core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -121,6 +122,9 @@ private FileIO io(String location) {
     String impl = implFromLocation(location);
     FileIO io = ioInstances.get(impl);
     if (io != null) {
+      if (io instanceof HadoopFileIO && ((HadoopFileIO) io).conf() == null) {
```

Comment:
Why does this need to change IO classes?

Reply:
Sorry, forgot to mention why I added this. It appears that the underlying `conf` of `HadoopFileIO` wasn't initialized when calling `StaticTableOperations.current()`:

![hadoop_io_conf_missing](https://user-images.githubusercontent.com/271029/234508649-e0499213-3992-4474-9b93-838314586839.png)